### PR TITLE
Add Tests to `scrapy check` command

### LINKED ISSUE #4222: SCRAPY_CHECK is not set while running contract
### Description

Hi, it seems that #3739 is not doing what the [documentation describe](https://docs.scrapy.org/en/latest/topics/contracts.html#detecting-check-runs):

`os.environ.get('SCRAPY_CHECK')` is returning `None` in my contract check.

### Steps to Reproduce

1. Create a project from scratch
2. Add a random spider
3. Contract code is as follow
```
    def parse(self, response):
        """
        @url http://www.amazon.com/s?field-keywords=selfish+gene
        @returns requests 1 1
        """
        print("test", os.environ.get('SCRAPY_CHECK'))
        if os.environ.get('SCRAPY_CHECK'):
            yield scrapy.Request(url="next_url")
```

**Expected behavior:** Request should be yielded as per the documentation

**Actual behavior:** Nothing happen

**Reproduces how often:** In my local project and with fresh project

### Versions

Windows
```
(globenv) C:\Users\johnl>scrapy version --verbose
Scrapy       : 1.8.0
lxml         : 4.4.1.0
libxml2      : 2.9.5
cssselect    : 1.1.0
parsel       : 1.5.2
w3lib        : 1.21.0
Twisted      : 19.10.0
Python       : 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 19:29:22) [MSC v.1916 32 bit (Intel)]
pyOpenSSL    : 19.0.0 (OpenSSL 1.1.1c  28 May 2019)
cryptography : 2.7
Platform     : Windows-10-10.0.18362-SP0
```

Linux
```
scrapy version --verbose
Scrapy       : 1.8.0
lxml         : 4.4.1.0
libxml2      : 2.9.9
cssselect    : 1.1.0
parsel       : 1.5.2
w3lib        : 1.21.0
Twisted      : 19.7.0
Python       : 3.6.8 (default, Oct  7 2019, 12:59:55) - [GCC 8.3.0]
pyOpenSSL    : 19.0.0 (OpenSSL 1.1.1d  10 Sep 2019)
cryptography : 2.8
Platform     : Linux-4.4.0-18362-Microsoft-x86_64-with-Ubuntu-18.04-bionic
```

### PR DESCRIPTION
**Pull request by ajaymittur28**
Continuing from PR #4298 which fixes #4222, added some basic tests for the `scrapy check` command as suggested by @Gallaecio and @elacuesta in the [original issue](https://github.com/scrapy/scrapy/issues/4222#issuecomment-598373876) and [PR](https://github.com/scrapy/scrapy/pull/4298#issuecomment-581161413). Would like some feedback on the tests!

### CODE REVIEW

**Comment by Gallaecio** [comment]
Nice. Could you resolve the conflict?

**Comment by ajaymittur28** [comment]
Do you think this should be enough for now or am I missing anything?

File: pytest.ini [hunk 1]
```diff
@@ -39,4 +39,4 @@ flake8-ignore =
     scrapy/utils/markup.py F403
     scrapy/utils/multipart.py F403
     scrapy/utils/url.py F403 F405
-    tests/test_loader.py E741
+    tests/test_loader.py E741
```
Comment (Gallaecio) [inline pytest.ini:42]: Please, remove the change here (you removed the empty line at the end of the file, we keep an empty line at the end of each file by convention).

Comment (ajaymittur28) [reply pytest.ini:42]: Sorry, must have removed it when resolving the merge conflict.

**Comment by wRAR** [comment]
Thanks!

**Pull request merged**